programs/bubblegum/program/src/processor/create_tree.rs

```diff
@@ -54,3 +70,40 @@ pub(crate) fn create_tree(
     );
     spl_account_compression::cpi::init_empty_merkle_tree(cpi_ctx, max_depth, max_buffer_size)
 }
+
+fn check_canopy_size(
+    ctx: &Context<CreateTree>,
+    max_depth: u32,
+    max_buffer_size: u32,
+) -> Result<()> {
+    let merkle_tree_bytes = ctx.accounts.merkle_tree.data.borrow();
+
+    let (header_bytes, rest) = merkle_tree_bytes.split_at(CONCURRENT_MERKLE_TREE_HEADER_SIZE_V1);
+
+    let mut header = ConcurrentMerkleTreeHeader::try_from_slice(header_bytes)?;
+    header.initialize(
+        max_depth,
+        max_buffer_size,
+        &ctx.accounts.tree_authority.key(),
+        Clock::get()?.slot,
+    );
+
+    let merkle_tree_size = merkle_tree_get_size(&header)?;
+
+    let (_tree_bytes, canopy_bytes) = rest.split_at(merkle_tree_size);
+
+    let required_canopy = if max_depth > MAX_ACC_PROOFS_SIZE {
+        max_depth - MAX_ACC_PROOFS_SIZE
+    } else {
+        0
+    };
```

Comment:
Note: This is fine and its readable but wanted to let you know you could use `saturating_sub` to make this a single line.